src/providers/tpm_provider/utils.rs

```diff
@@ -84,3 +91,134 @@ pub struct PasswordContext {
     pub context: TpmsContext,
     pub auth_value: Vec<u8>,
 }
+
+pub fn parsec_to_tpm_params(attributes: KeyAttributes) -> Result<KeyParams> {
+    match attributes.key_type {
+        KeyType::RsaKeyPair => {
+            let size = match attributes.key_bits {
+                x @ 1024 | x @ 2048 | x @ 3072 | x @ 4096 => x.try_into().unwrap(), // will not fail on the matched values
+                _ => return Err(ResponseStatus::PsaErrorInvalidArgument),
+            };
+            Ok(KeyParams::Rsa {
+                size,
+                scheme: convert_asym_scheme_to_tpm(attributes.key_policy.key_algorithm)?,
+                pub_exponent: 0,
+            })
+        }
+        KeyType::EccKeyPair { curve_family } => Ok(KeyParams::Ecc {
+            scheme: convert_asym_scheme_to_tpm(attributes.key_policy.key_algorithm)?,
+            curve: convert_curve_to_tpm(curve_family, attributes.key_bits)?,
+        }),
+        _ => Err(ResponseStatus::PsaErrorNotSupported),
+    }
+}
+
+fn convert_asym_scheme_to_tpm(algorithm: Algorithm) -> Result<AsymSchemeUnion> {
```

Comment:
Would it be better to take `AsymmetricSignature` as parameter here? Or rename the function `convert_scheme_to_tpm` if that still makes sense for other algorithms.

Reply:
Yes and no - asymmetric encryption schemes exist, it's just that at the moment we don't deal with them. I could split this further into multiple methods, for each kind of asymmetric scheme, but I don't think it's wrong to take an algorithm here - especially since pattern matching in Rust allows things like below. Otherwise you'd have two distinct matches doing the same work.

Reply:
Got it, I forgot about asymmetric encryption